server/graphql/v2/mutation/TransactionMutations.ts

```diff
@@ -64,19 +71,43 @@ const transactionMutations = {
         }
         refundedTransaction = await fetchTransactionWithReference({ legacyId: transaction.RefundTransactionId });
       }
-      const orderToUpdate = await models.Order.update(
-        {
-          status: orderStatus.REJECTED,
-        },
-        {
-          where: { id: refundedTransaction.OrderId },
-          returning: true,
-        },
-      );
+
+      const orderToUpdate = await models.Order.findOne({
+        where: { id: refundedTransaction.OrderId },
+        include: modelArray,
+      });
+
       if (!orderToUpdate) {
         throw new NotFound('Order not found');
       }
 
+      if (orderToUpdate.SubscriptionId) {
+        if (!req.remoteUser.isAdmin(orderToUpdate.FromCollectiveId)) {
+          throw new Unauthorized("You don't have permission to cancel this recurring contribution");
+        }
+        if (!orderToUpdate.Subscription.isActive && orderToUpdate.status === orderStatus.CANCELLED) {
+          throw new Error('Recurring contribution already canceled');
+        }
+        await orderToUpdate.update({ status: orderStatus.REJECTED });
+        await orderToUpdate.Subscription.deactivate();
+        await models.Activity.create({
```

Comment:
If we don't use this Activity for something, we can just remove it.

Reply:
If we include it, it should be REJECTED, not CANCELLED

Reply:
Don't we have to create the `Activity` for the subscription being cancelled so that is recorded in the history?